### PR TITLE
Add Gitpod Support

### PR DESCRIPTION
**Pull request by dhc02**
I spent some time getting gitpod.io to work on my fork and thought I'd create a pull request in case it's something you'd like to add to the official repo.

If you're not familiar, gitpod.io is a service that lets you spin up a browser-based IDE for different contexts. That's sort of a nebulous description, but it's a nebulous service. It's very powerful. It's free for personal use by default, and provides a fully-functioning VM including access to docker and sudo, and rust is a first-class citizen. 

You can try it with any github repo by prepending `gitpod.io/#` to any github URL. It would be instructive to take a look at what happens when you do it using my fork: https://gitpod.io/#https://github.com/dhc02/IRust

What this pull request represents is the addition of a config file and two simple scripts to the repo. WIth that config file added, gitpod.io now understands how to treat the iRust repo and make it functional. All this means is that in a "prebuild" step that takes place after any commit to the repo, iRust, racer, cargo-edit, and cargo-asm are installed on top of their base docker container. Then, when you spin up a gitpod iRust workspace, you don't have to wait for these steps to complete and can begin using iRust immediately.

TL:DR; This allows you to provide people with a browser-based iRust IDE environment to play around in, and every time you commit a change to the iRust repo, those changes are automatically integrated into that IDE (after about 6 minutes of compiling in the prebuild task).

### CODE REVIEW

**Comment by sigmaSd** [comment]
Thanks for taking the time to fill this pr, I'll test soon hopefully

Just one note I prefer those scripts inlined, so I prefer to revert the last commit

**Comment by sigmaSd** [comment]
I tested it and I like it, irust is even feels fast

Some notes: 
- As I said before I prefer  inline scripts (to keep the root repo relatively clean)
- The rust extension should be not be used (its deprecated), I think rust-analyzer is on by default?
- Adding a note to the README would be great, something like "Try it with not setup (via gitpod) [link]", I think under the first paragraph
- One nitpick I prefer to write irust as IRust (like IPython)

One other note to self:
- Make `:help` output a more compact version, and add `:help full` for the current behavior

Thanks again!

**Comment by sigmaSd** [comment]
One other note:
- gitpod doesn't have `.config`  directory by default, so this https://github.com/sigmaSd/IRust/blob/master/crates/irust/src/irust/options.rs#L139 needs to change to `create_dir_all`, I think its an ok change anyway

**Comment by dhc02** [comment]
I've added a commit addressing all of your feedback. 

I went ahead and made the change in `crates/irust/src/irust/options.rs`. Let me know if you'd rather leave that for a future update.

One important note: Right now, the `github: prebuilds:` config lines mean that when the `dhc02/irust`repo is updated, the prebuild starts recompiling IRust, racer, cargo-edit, and cargo-asm. But this functionality depends on me having installed the [Gitpod "app"](https://github.com/apps/gitpod-io) on my Github account. If you integrate this pull request into your repo and you want the same automatic functionality, you'll need to install the same app (or set up manual webhooks for certain actions on the repo).

One very cool thing about Gitpod that I just found out is that if you have the chrome extension installed, then the `Gitpod` button appears on the pull request page, and if you click it, it opens the repo in the context of the pull request as it sits right now, which makes amending pull requests almost foolproof.

**Comment by sigmaSd** [comment]
@dhc02 That looks good, Thanks a lot!

**Comment by sigmaSd** [comment]
Thanks for the info about gitpod app, I think that's the most important feature I want from adding gitpod so I went ahead and installed it
